### PR TITLE
fix(Core): Fixing item enchants randomization

### PR DESCRIPTION
**Pull request by wizzymore**
##### CHANGES PROPOSED:
-  Randomize weapons enchants from all sources

~This can be flagged as WIP for the moment, as it only fixes .send items command, not all the sources~

##### ISSUES ADDRESSED:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/2324

##### TESTS PERFORMED:
- Tested .send items command
- Tested items with enchants from vendors

Images can be found [here](https://imgur.com/a/4CttV5z)
 

##### KNOWN ISSUES AND TODO LIST:

- [x] Randomize enchants for items send through mail
- [x] Randomize enchants for items bought from a vendor

##### Target branch(es):
- [x] Master

 

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by wizzymore** [comment]
If anyone can help me figure out where in the core the items from vendors are added to inventory that would be awesome..

**Comment by wizzymore** [comment]
@FrancescoBorzi @Stoabrogga please take a look at this

@Shard-MW help me by testing this

File: src/server/scripts/Commands/cs_misc.cpp [hunk 1]
```diff
@@ -2743,6 +2743,7 @@ class misc_commandscript : public CommandScript
         {
             if (Item* item = Item::CreateItem(itr->first, itr->second, handler->GetSession() ? handler->GetSession()->GetPlayer() : 0))
             {
+            	item->SetItemRandomProperties(Item::GenerateItemRandomPropertyId(itr->first));
```
Comment (Winfidonarleyan) [inline src/server/scripts/Commands/cs_misc.cpp]: add whitespace

**Comment by Knindzagxg** [comment]
Works pretty good for me, nice one.

**Comment by Shard-MW** [comment]
Works fine for me too :)
I had to push my PR too but just came home, so I'm putting it into a compare to show you because it's actually not the same fix at all haha

I tested  
- [ ] Add the item through command (.add 727)
- [ ] Send the item by mail through command (.send items Player "text" "text" 727:2)
- [ ] Loot item 727 from creature
- [ ] Loot item 727 from creature (in group)
- [ ] Loot item 727 from creature (in raid)
- [ ] Put item in AH and bought it from another player
- [ ] Trade item to another player
- [ ] Put the item in bank and bring it back
- [ ] Put item in guild bank and bring it back with another player
- [ ] Buy the item from vendor
- [ ] Drag and drop the item from vendor list to equip slot

and it's working for Wizzy's PR too

**Comment by wizzymore** [comment]
For the better solve of this problem check this PR: https://github.com/azerothcore/azerothcore-wotlk/pull/2331

**Pull request closed without merging**